src/System.CommandLine/StringExtensions.cs

```diff
@@ -210,6 +200,74 @@ void ReadResponseFile(string filePath, int i)
             }
         }
 
+        private static List<string> NormalizeRootCommand(
+            CommandLineConfiguration commandLineConfiguration, 
+            IReadOnlyList<string> args)
+        {
+            if (args == null)
+            {
+                args = new List<string>();
+            }
+
+            string potentialRootCommand = null;
+
+            if (args.Count > 0)
+            {
+                try
+                {
+                    potentialRootCommand = Path.GetFileName(args[0]);
+                }
+                catch (ArgumentException)
+                {
+                    // possible exception for illegal characters in path on .NET Framework
+                }
+
+                if (commandLineConfiguration.RootCommand.HasRawAlias(potentialRootCommand))
+                {
+                    return args.ToList();
+                }
+            }
+
+            var commandName = commandLineConfiguration.RootCommand.Name;
+
+            if (FirstArgMatchesRootCommand())
+            {
+                args = new[]
+                       {
+                           commandName
+                       }.Concat(args.Skip(1)).ToList();
+            }
+            else
+            {
+                args = new[]
+                       {
+                           commandName
+                       }.Concat(args).ToList();
+            }
+
+            return args.ToList();
```

Comment:
Both branches of the above if statement call `.ToList()`. You can likely remove them in favor of just this one.